example-plugins/patient_portal_header_widget/handlers/header_widget_handler.py

```diff
@@ -0,0 +1,33 @@
+from canvas_sdk.effects import Effect
+from canvas_sdk.effects.widgets import PortalWidget
+from canvas_sdk.events import EventType
+from canvas_sdk.handlers.base import BaseHandler
+from canvas_sdk.templates import render_to_string
+from canvas_sdk.v1.data import Patient
+
+
+# Inherit from BaseHandler to properly get registered for events
+class HeaderWidgetHandler(BaseHandler):
+    """Handler responsible for rendering a header widget in the patient portal upon receiving a configuration event."""
+
+    RESPONDS_TO = EventType.Name(EventType.PATIENT_PORTAL__WIDGET_CONFIGURATION)
+
+    def compute(self) -> list[Effect]:
+        """This method gets called when an event of the type RESPONDS_TO is fired."""
+        patient = Patient.objects.get(id=self.target)
```

Comment:
Since you're just using the patient preferred name, you could only fetch the fields needed for that property.

```suggestion
        patient = Patient.objects.only("first_name", "last_name", "suffix", "nickname").get(id=self.target)
```